src/Stratis.Bitcoin.Features.Consensus/Rules/CommonRules/CalculateStakeRule.cs

```diff
@@ -30,6 +30,15 @@ public override Task RunAsync(RuleContext context)
 
             context.NextWorkRequired = this.PosParent.StakeValidator.GetNextTargetRequired(this.PosParent.StakeChain, context.ValidationContext.ChainedHeader.Previous, context.Consensus, posRuleContext.BlockStake.IsProofOfStake());
 
+            BlockHeader header = context.ValidationContext.Block.Header;
+
+            // Check proof of work.
+            if (header.Bits != context.NextWorkRequired)
+            {
+                this.Logger.LogTrace("(-)[BAD_DIFF_BITS]");
+                ConsensusErrors.BadDiffBits.Throw();
```

Comment:
Now you added exception to the method but not comment for the exception. Also you could remove extra space in the high hash exception comment.